open_spiel/games/dark_chess_test.cc

```diff
@@ -0,0 +1,36 @@
+// Copyright 2019 DeepMind Technologies Ltd. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "open_spiel/spiel.h"
+#include "open_spiel/tests/basic_tests.h"
+
+namespace open_spiel {
+namespace dark_chess {
+namespace {
+
+namespace testing = open_spiel::testing;
+
+void BasicDarkChessTests() {
+  testing::LoadGameTest("dark_chess");
+  testing::NoChanceOutcomesTest(*LoadGame("dark_chess"));
+  testing::RandomSimTest(*LoadGame("dark_chess"), 100);
```

Comment:
Can you add a test for the chessboard where you generate some specific simple board where the king can / cannot move in / out of check, based on the flag set? Generate the legal actions and test whether the action is present there or not.

Reply:
?